DiscordChatExporter.Core/Exporting/JsonMessageWriter.cs

```diff
@@ -64,6 +64,31 @@ public JsonMessageWriter(Stream stream, ExportContext context)
         await _writer.FlushAsync(cancellationToken);
     }
 
+    private async ValueTask WriteReactionUserAsync(
+        User user,
+        CancellationToken cancellationToken = default)
+    {
+        // same as WriteUserAsync, but without discriminator, color and roles
+
+        _writer.WriteStartObject();
+
+        _writer.WriteString("id", user.Id.ToString());
+        _writer.WriteString("name", user.Name);
+        _writer.WriteString("nickname", Context.TryGetMember(user.Id)?.DisplayName ?? user.DisplayName);
+        _writer.WriteBoolean("isBot", user.IsBot);
+
+        _writer.WriteString(
+            "avatarUrl",
+            await Context.ResolveAssetUrlAsync(
+                Context.TryGetMember(user.Id)?.AvatarUrl ?? user.AvatarUrl,
+                cancellationToken
+            )
+        );
+
+        _writer.WriteEndObject();
+        await _writer.FlushAsync(cancellationToken);
+    }
```

Comment:
If this method is only used in one place, then you can just inline it. Make sure to add a comment explaining that it only writes a limited version of the user.